sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobStorageResourceContainerOptions.cs

```diff
@@ -10,12 +10,23 @@ namespace Azure.Storage.DataMovement.Blobs
     /// </summary>
     public class BlobStorageResourceContainerOptions
     {
+        private BlobType _blobType = default;
+        internal bool _isBlobTypeSet = false;
+
         /// <summary>
         /// Optional. The <see cref="Storage.Blobs.Models.BlobType"/> that will be used when uploading blobs to the destination.
         ///
         /// Defaults to preserving the blob type if the source is also a blob. If the source is not a blob, will default to Block Blob.
         /// </summary>
-        public DataTransferProperty<BlobType?> BlobType { get; set; }
+        public BlobType? BlobType
+        {
+            get => _blobType;
+            set
+            {
+                _blobType = value ?? default;
```

Comment:
Do we need the `?? default`?

Reply:
Ah it's cause I made the internal type for some reason not nullable. Changed the top type to nullable and removed default here.